es6/test/lib/createRouteHandler.js

```diff
@@ -0,0 +1,339 @@
+'use strict';
+//jshint camelcase: false
+//jshint maxstatements: false
+
+var assert = require('assert');
+var Bluebird = require('bluebird');
+var sinon = require('sinon');
+
+describe('createRouteHandler', function() {
```

Comment:
I'm not going to continue to comment on the tests. You should take a minute to rewrite them in es6 syntax.